lib/util.js

```diff
@@ -0,0 +1,170 @@
+'use strict'
+
+const { URL } = require('url')
+const Ref = require('json-schema-resolver')
+
+function addHook (fastify) {
+  const routes = []
+  const sharedSchemasMap = new Map()
+
+  fastify.addHook('onRoute', (routeOptions) => {
+    routes.push(routeOptions)
+  })
+
+  fastify.addHook('onRegister', async (instance) => {
+    // we need to wait the ready event to get all the .getSchemas()
+    // otherwise it will be empty
+    instance.addHook('onReady', (done) => {
+      const allSchemas = instance.getSchemas()
+      for (const schemaId of Object.keys(allSchemas)) {
+        if (!sharedSchemasMap.has(schemaId)) {
+          sharedSchemasMap.set(schemaId, allSchemas[schemaId])
+        }
+      }
+      done()
+    })
+  })
+
+  return {
+    routes,
+    Ref () {
+      const externalSchemas = Array.from(sharedSchemasMap.values())
+      return Ref({ clone: true, applicationUri: 'todo.com', externalSchemas })
+    }
+  }
+}
+
+// The swagger standard does not accept the url param with ':'
+// so '/user/:id' is not valid.
+// This function converts the url in a swagger compliant url string
+// => '/user/{id}'
+function formatParamUrl (url) {
+  let start = url.indexOf('/:')
+  if (start === -1) return url
+
+  const end = url.indexOf('/', ++start)
+
+  if (end === -1) {
+    return url.slice(0, start) + '{' + url.slice(++start) + '}'
+  } else {
+    return formatParamUrl(url.slice(0, start) + '{' + url.slice(++start, end) + '}' + url.slice(end))
+  }
+}
+
+function consumesFormOnly (schema) {
```

Comment:
this is swagger-specific, likely would be helpful to separate OpenAPI stuff from Swagger stuff and from common stuff.